### DISCUSSION TITLE
CDC and Audio: `TU_ASSERT(packet_sz_tx_max <= audio->ep_in_sz);` after USB re-attach

### DISCUSSION THREAD

**Discussion by ra1nb0w** [hathach/tinyusb#2724] [Q&A]
Hi,
I have implemented tinyUSB in a custom STM32H563 board with Audio, CDC and DFU Runtime following the headset example.
The first USB connection works fine. If I disconnect the USB cable and re-attach it I always get in the debugger the following assert `TU_ASSERT(packet_sz_tx_max <= audio->ep_in_sz);`. `audiod_calc_tx_packet_sz()` is called with the end-point equals to 0.
The full trace can be found in the file [debug.txt](https://github.com/user-attachments/files/16286842/tinyusb_trace.txt)

Suggestions on how to trace the issue?

Thank you very much

**Comment by HiFiPhile** [comment]
Good spot, there are some variables not reset to 0 on driver reset, please try: https://github.com/HiFiPhile/tinyusb/tree/uac_reset

**Comment by ra1nb0w** [comment]
Amazing. It fixes the issue! Thank you very much.
